### ISSUE TITLE
Wrong definition for IL_Add()

### ISSUE THREAD

**Issue by visua0** [GroggyOtter/ahkv2_definition_rewrite#4]
Current definition:
```
IL_Add(ImageListID, IconOrPic, Num, Resize) => Integer
```
From the docs:
```
IconIndex := IL_Add(ImageListID, IconFileName [, IconNumber])
IconIndex := IL_Add(ImageListID, PicFileName, MaskColor, Resize)
```

**Comment by GroggyOtter** [comment]
Like many other functions, the parameters of this function have been merged and rewritten for my definition file.  
Please read the calltip.  
It clearly covers how to use the function correctly and notes the parameter differences between both icons and pics.

**Comment by visua0** [comment]
> Like many other functions, the parameters of this function have been merged and rewritten for my definition file.

Sorry I'm not entirely sure what this means.

Yes the calltip accurately describes that the meaning of `Num` depends on the value of `IconOrPic`, but neither the definition nor the calltip mentions that `Num` and `Resize` are optional (unless `IconOrPic` is a picture).
In fact both examples from the calltip result in problems due to this:
![image](https://github.com/GroggyOtter/ahkv2_definition_rewrite/assets/63650/a65326c4-b6fd-4b89-a279-4e86927261a6)

**Comment by GroggyOtter** [comment]
Thanks for taking time to report that.  
I think I understand what you mean.  

I've updated the method.  
This should no longer mark it as an error with 3 parameters.  
I also updated the wording of the calltip to be more clear.

![Code_v49pCe6Oqn](https://github.com/GroggyOtter/ahkv2_definition_rewrite/assets/29220773/c34d7b72-2299-4d9c-ae3d-ef63e6f6d7cf)

This will be in the next update.  
If you want an early patch, replace the current `IL_Add()` method with this one:

```
/**
 * @description {@link https://www.autohotkey.com/docs/v2/lib/ListView.htm#IL_Add|`IL_Add()`}  
 * Adds an icon or picture to the specified ImageList.  
 * @param {(Integer)} ImageListID  
 * The identifying number of an image list generated by {@link https://www.autohotkey.com/docs/v2/lib/ListView.htm#IL_Create|IL_Create()}.
 * @param {(String)} IconOrPicFile  
 * Filename to an icon or picture file  
 * - `Icon` = .ICO .CUR .ANI .EXE .DLL .CPL .SCR {@link https://www.autohotkey.com/docs/v2/misc/ImageHandles.htm|Icon Handle "HICON:"} or another icon resource
 * - `Pic` = .BMP .GIF .JPG .PNG .TIF .Exif .WMF .EMF or a {@link https://www.autohotkey.com/docs/v2/misc/ImageHandles.htm|Bitmap Handle "HBITMAP:"}  
 * @param {(Integer)} Num  
 * Dependent on Icon or Picture.
 * - `Icon` = Num is the icon number to use  
 *   If this is no sign, the number indicates the icon group to use  
 *   A negative sign indicates the number is used as a resource ID for an icon inside an executable  
 *   Num can be omitted when using an icon and will default to `1`.  
 * - `Picture` = A transparency mask color in hex  
 *   The color white `0xFFFFFF` works well for most pictures  
 * @param {(Boolean)} Resize  
 * This parameter only applies to pictures, not to icons.
 * - `1` = Picture is scaled to become a single icon  
 * - `0` = Picture is divided into a number of icons that fit inside the given width  
 * @returns {(Integer)} - On success, the ImageList index position the icon was stored in is returned, otherwise `0`.
 * @example <caption></caption>  
 * IL_Add(ImageListID, 'C:\Path\to\application.exe', 2)
 * IL_Add(ImageListID, 'C:\Path\to\image.bmp', 0xFFFFFF, )
 */
IL_Add(ImageListID, IconOrPicFile [, Num:=1, Resize]) => Integer
```

**Comment by GroggyOtter** [comment]
Fixed with v1.3
